src/route/user/index.ts

```diff
@@ -68,8 +68,26 @@ const user: FastifyPluginCallback = (app, _, done) => {
   if (wclStrategy) {
     passport.use(wclStrategy);
     // TODO: This should redirect to separate WCL login page
-    app.get("/login/wcl", passport.authenticate("wcl"));
-    app.get("/login/wcl/callback", passport.authenticate("wcl", options));
+    app.get("/login/wcl", (req, res) => {
+      req.session.set("returnTo", req.query.redirect);
+      passport.authenticate("wcl")(req, res);
+    });
+    app.get("/login/wcl/callback", (req, reply) => {
+      passport.authenticate("wcl", async (_Request, _Reply, err, User) => {
+        if (err || !User) {
+          return reply.redirect(options.failureRedirect);
+        }
+
+        // Get return url before logging in the user or it will be lost
+        const returnToUrl = req.session.get("returnTo")
+          ? `http://localhost:3000${req.session.get("returnTo")}`
```

Comment:
this shouldn't hardcode the localhost address (clearly :P). should enforce that it is a root redirect (by prepending `/` if it doesn't start with `/`) so there are no shenanigans, but otherwise can leave the TLD off.